twap/main.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/blocky/as-demo/as"
+	"github.com/blocky/as-demo/price"
+)
+
+type ArgsIterate struct {
+	TokenAddress string          `json:"token_address"`
+	ChainID      string          `json:"chain_id"`
+	NumSamples   int             `json:"num_samples"`
+	EAttest      json.RawMessage `json:"eAttest"`
+	TAttest      json.RawMessage `json:"tAttest"`
+	Whitelist    json.RawMessage `json:"whitelist"`
+}
+
+type ArgsTWAP struct {
+	EAttest   json.RawMessage `json:"eAttest"`
+	TAttest   json.RawMessage `json:"tAttest"`
+	Whitelist json.RawMessage `json:"whitelist"`
+}
+
+type Result struct {
+	Success bool
+	Error   string
+}
+
+type PriceSamples []price.Price
```

Comment:
It is not clear to me why we are creating this type.  Seems unnecessary.

Reply:
OK